src/librustc_infer/lib.rs

```diff
@@ -17,12 +16,10 @@
 #![feature(bool_to_option)]
 #![feature(box_patterns)]
 #![feature(box_syntax)]
-#![feature(drain_filter)]
 #![feature(never_type)]
 #![feature(range_is_empty)]
 #![feature(in_band_lifetimes)]
 #![feature(crate_visibility_modifier)]
-#![recursion_limit = "512"]
```

Comment:
This was only needed for the parallel compiler. Can you try building it before removing this?

Reply:
I managed to compile as is.

Reply:
Looks like this is needed anyway for `rustdoc`.